### PR TITLE
interpolate component of field to a component of another field

### PR DESCRIPTION
**Pull request by mwichro**
diffenet dof handler and different elements are possible.

### CODE REVIEW

File: include/deal.II/numerics/vector_tools_copy_to_different_dof_handler.h [hunk 1]
```diff
@@ -0,0 +1,335 @@
+/*
+ * copy_to_different_dof_handler.h
+ *
+ *  Created on: Apr 29, 2020
+ *      Author: mwichro
```
Comment (luca-heltai) [inline include/deal.II/numerics/vector_tools_copy_to_different_dof_handler.h:5]: Can you put the declaration in include/deal.II/numerics/vector_tools.h,  the implementation in include/deal.II/numerics/vector_tools.templates.h, and the instantiation in a source/numerics/vector_tools_interpolate_to_different_dof_handler.cc and source/numerics/vector_tools_interpolate_to_different_dof_handler.inst.in?

Comment (luca-heltai) [inline include/deal.II/numerics/vector_tools_copy_to_different_dof_handler.h:5]: Can you move the declaration to include/deal.II/numerics/vector_tools.h and the implementation to include/deal.II/numerics/vector_tools.templates.h?

File: include/deal.II/numerics/vector_tools_copy_to_different_dof_handler.h [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    template <int dim,
+              int spacedim,
+              typename VectorType,
+              template <int, int> class DoFHandlerType>
+    void
+    interpolate_to_different_dof_handler(
+      VectorType &                                      dst,
+      const DoFHandlerType<dim, spacedim> &             dof_handler,
+      const unsigned int &                              dst_component,
+      const LinearAlgebra::distributed::Vector<double> &src,
```
Comment (luca-heltai) [inline include/deal.II/numerics/vector_tools_copy_to_different_dof_handler.h:36]: Can you template also on the input vector type?

File: include/deal.II/numerics/vector_tools_copy_to_different_dof_handler.h [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+// apply_transform is defined here
+// I am probably including wrong file, so please check this
+#include <deal.II/numerics/vector_tools_interpolate.templates.h>
+
+namespace dealii
+{
+  namespace VectorTools
+  {
+    // Modyfied internal interpolate
+    //
```
Comment (luca-heltai) [inline include/deal.II/numerics/vector_tools_copy_to_different_dof_handler.h:26]: Can you copy the documentation here, modifying what is different from the other version?

File: include/deal.II/numerics/vector_tools_copy_to_different_dof_handler.h (hunk 1 above)
Comment (masterleinad) [reply include/deal.II/numerics/vector_tools_copy_to_different_dof_handler.h:5]: You mean `vector_tools_interpolate.h` and `vector_tools_interpolate.templates.h`. :slightly_smiling_face:

**Comment by drwells** [comment]
@mwichro would you like to finish this PR? If not lets consider finishing this ourselves.

**Pull request closed without merging**